### ISSUE TITLE
Strange mod interactions with 22d

### ISSUE THREAD

**Issue by reik019** [ReikaKalseki/Reika_Mods_Issues#2187]
Like many people here reported, release 22 causes crashes with MineFactory Reloaded, but the strangest thing is, that when I load Tinker's Construct, the game loads just fine. 
It might be some interface overriding another, perhaps?
I'm not a technical one, but if you need the modlist or logs/crashlogs I can provide them on pastebin.

**Comment by ReikaKalseki** [comment]
I need a crash log, yes.

**Comment by reik019** [comment]
The typical crash with MineFactory Reloaded: [https://pastebin.com/Tgf4K5dt](https://pastebin.com/Tgf4K5dt)
But it launches fine if I add Tinker's construct alongside MFR: [https://pastebin.com/99QJ703q](https://pastebin.com/99QJ703q)

Modlist (first run, the one that crashed):
CoFHCore-[1.7.10]3.1.4-329
DragonAPI+1.7.10+V22d
ElectriCraft+1.7.10+V22a
MineFactoryReloaded-[1.7.10]2.8.1-174
ReactorCraft+1.7.10+V22a
RotaryCraft+1.7.10+V22d
ThermalDynamics-[1.7.10]1.2.1-172
ThermalExpansion-[1.7.10]4.1.5-248
ThermalFoundation-[1.7.10]1.2.6-118

Modlist (second run, the one that launched fine):
CoFHCore-[1.7.10]3.1.4-329
DragonAPI+1.7.10+V22d
ElectriCraft+1.7.10+V22a
Mantle-1.7.10-0.3.2b
MineFactoryReloaded-[1.7.10]2.8.1-174
ReactorCraft+1.7.10+V22a
RotaryCraft+1.7.10+V22d
TConstruct-1.7.10-1.8.8
ThermalDynamics-[1.7.10]1.2.1-172
ThermalExpansion-[1.7.10]4.1.5-248
ThermalFoundation-[1.7.10]1.2.6-118

**Comment by OvermindDL1** [comment]
Only thing I really see that would change that is that MFR has a dependency on TConstruct if it exists via `after:TConstruct`.  Why is TConstruct loaded in relation to MFR?  When is MFR loaded in relation to reika's mods?  Probably a loading order issue somewhere.

**Comment by ReikaKalseki** [comment]
No, this is an API version issue, specifically on that one class. I reference that class but do not package it, so this goes beyond simple old-version-inclusion.

**Comment by OvermindDL1** [comment]
@ReikaKalseki It shouldn't be packaged anyway, but still, this file exists in MFR at `powercrystals.minefactoryreloaded.item.tool.ItemFactoryHammer`.  And it does *not* exist in TConstruct.

**Comment by ReikaKalseki** [comment]
MFR is obviously going to package its own API file...

**Comment by OvermindDL1** [comment]
> MFR is obviously going to package its own API file...

Yes?  And?  That's what I said, MFR packages it, TConstruct does not, Reika's Mods do not, ergo it's not a conflicting package issue among those mods.  What are you referencing with your `MFR is obviously going to package its own API file...` post I'm curious?

It could still be caused by another mod if any others are installed, the OP had a lot of ThermalMods, so who knows what they affect.

**Comment by ReikaKalseki** [comment]
> What are you referencing with your MFR is obviously going to package its own API file... post I'm curious?

Your phrasing (`It shouldn't be packaged anyway, but still, this file exists in MFR`) implied that this is unusual or problematic.

**Comment by OvermindDL1** [comment]
> Your phrasing (It shouldn't be packaged anyway, but still, this file exists in MFR) implied that this is unusual or problematic.

Ah, nope, just confirming to the OP that the issue doesn't lay in those locations.

**Comment by reik019** [comment]
To be fair, the Thermal Series doesnt do anything like ASM patching, according to the logs, the one that does that its the coremod of Thermal and MFR itself.

@ReikaKalseki some thing that might be affecting would be that MFR Requires Thermal Expansion to run alongside TCon, removing Thermal Expansion + Company results in a crash, unrelated to DragonAPI/RoC, ReC or ElC.

Also @OvermindDL1 just accept your mistakes before you get ridiculed by your own words, you never know if there's a smartass near, or at least take some more time to see if your responses make sense before you post,
